perceval/converters/abstract_converter.py

```diff
@@ -70,11 +74,15 @@ def _configure_processor(self, gate_circuit, **kwargs):
         """
         qname = kwargs.get("qname", "Q")  # Default value, set any name provided by the gate circuit
         n_qbits = self.count_qubits(gate_circuit)
+
+        qubit_names = kwargs.get(
```

Comment:
The previous code assumed that there was a single qubit array, so it received its name in the `qname` (for example `q`) and created the list of name `q[0]`, `q[1]` and so on.

But cQASM supports individual names mixed with arrays, we can have `a`, `b`, `q[0]`, `q[1]` so we need a way of passing the full list of names. The proposed code is compatible with both passing a single array name (for the existing codebase), and the full explicit list.

Reply:
Ok, I would like to check it a bit. Do you have some example code I could use to reproduce?

Reply:
Maybe I did not explain it well!

***

Here's what the qiskit converter does:

```
qubit_names = qc.qregs[0].name
self._configure_processor(qc, qname=qubit_names)
```

So if the first qreg is named `z` and there are 4 qubits, the ports created in `_configure_processor` will be named z0, z1, z2, z3.

***

What the myqlm converter does:

```
self._configure_processor(qlmc)
```

We do not provide a qname here, so the default `Q` is assumed and the qubits are called Q0, Q1, Q2...

***

But cQASM allows you to name individual qubits as if they were variables:

```
version 3
qubit control
qubit[2] target
CNOT control, target[2]
```

So in order to preserve the names mentioned in the source, I need to pass the full list of names, instead of using a single string suffixed by the index.

There are thus 3 ways of naming qubits:
* Do not pass anything: qubits are implicitly named `Q0`, `Q1`, `Q2`
* Prefix: pass as string in `qname` and they'll be named qname0, qname1, qname2
* Pass an explicit list of names in `qubit_names`.

See the test I have added in `test_converter_qubit_names` . I think it's a good idea to preserve the naming of the qubits from the source file.

Reply:
Thanks for the information.